### DISCUSSION TITLE
PS2 OPL SMB settings

### DISCUSSION THREAD

**Discussion by danmons** [retronas/discussions#63] [Q&A]
Starting a discussion item specifically around PlayStation 2 running OpenPS2Loader and loading games over SMB1 from Samba, and compatibility of specific titles. 

The first thing I ask anyone joining the discussion is please try both the latest stable released and the latest beta release of OPL, downloadable from here:
https://github.com/ps2homebrew/Open-PS2-Loader/releases

Ancient versions of OPL, or versions of HDLoader will be ignored, as there are constant improvements to the upstream project. 

Also worth checking OPL issues for known bugs:
https://github.com/ps2homebrew/Open-PS2-Loader/issues

Following on from there, please list problem games in here, and please be specific.  Note the exact version, territory and release of the game and be as specific as possible with the problem description (e.g.: if there's video or audio skipping, not where this happens and what the user has to do to reproduce the problem).

If possible, please run checksums on your ISO and verify against Redump's database, and then link to the exact title.  The Redump database is here, and linking to the exact game helps testers make sure they have the same version as you when testing:
http://redump.org/discs/system/ps2/

Redump list CRC32, MD5 and SHA-1 checksums for every game that you can verify on your ISO.

Each game has several options you can tweak.  Jump to timestamp 23:16 in this video to see the settings page, and the various options you can set.  Try modifying these, and if you find a combination for a specific game that solves your issue, please let us know in this discussion:

https://youtu.be/c3AldDYZ8_k?t=1396

**Comment by sairuk** [comment]
I think being explicit about an actual >minver of OPL that will be considered is ideal in this scenario, help users identify where they are at before they contribute and also include the version they are running.

I think we should also exhibit how to gather the information we want to make this thread useful, doesn't matter if its a 3rd party video but something that demonstrates how to validate version of opl, checksums against redump and _at a minimum how to collect the disc_id from an iso, does OPL have that available onscreen etc?

Do we provide a script people can run with Sabre to pull the redump ps2 dat in and scan iso ondevice now that we have Sabre available?

**Reply by danmons** [reply]
> Do we provide a script people can run with Sabre to pull the redump ps2 dat in and scan iso ondevice now that we have Sabre available?

This sounds really useful.

**Reply by sairuk** [reply]
Only caveat is we'd ~~need to~~ should point people to MPF dump their own isos, which in theory we can prob integrate into retronas (on ideas iirc) and then throw some USB disc drives at it to test, I don't know that I have any lying about to test with. Let alone they need to be very specific drives.

**Comment by uv7jWsX7qO** [comment]
This is probably a very silly question. I followed the basic setup guides and can get my PS2 to see the network share and the isos on it. However, when I try to actually play any games I get the stuck at the pink debug screen.

I should note that my setup only uses a network switch and not a router. Is that the cause of my issue or can it work just fine as an internal network running off of a network switch?

Reference Info:
OPL Version 1.0
ISO SHA1: 0f86e6141d8c352f04cb78cea196b46b3a0d45cb
Redump: http://redump.org/disc/3563/

**Reply by sairuk** [reply]
It will run fine off of a basic gigabit switch, if its 10/100 you _may_ have issues. 

Your OPL version is old so that is likely the problem, please update to 1.1.0 as per our [wiki](https://github.com/danmons/retronas/wiki/OpenPS2Loader)

**Reply by uv7jWsX7qO** [reply]
I'll try updating but it is a 10/100 switch that I had lying around. My rationale was that the ps2 ports weren't gigabit so it would suffice but I don't mind picking up a cheap router with a gigabit switch if that would be better.

**Reply by sairuk** [reply]
I'd start with the OPL update, latest version should help compatibility.

I agree the port speed of the ps2 and switch requirements thereafter is a reasonable assumption imho particularly if you're only doing ps2.

**Reply by uv7jWsX7qO** [reply]
I just updated it and I get the same result. Given net cost right now is 0, a little bit for a new router/switch is not a huge deal. I'll probably end up getting a wireless router since I plan on expanding to Dreamcast when I add the MODE into that.

The only other thing I can think of is if it's the hard drive, it's a usb 3 WD passport so not the fastest thing in the world but should be good enough.

**Reply by uv7jWsX7qO** [reply]
So still not sure on what I'm doing wrong here. Swapped the switch for a gigabit switch and checked my settings per the guide but still just getting stuck at the pink debug screen.

Could it be the hard drive is not up to the task? I'm quite confused since the PS2 definitely sees the share but just can't seem to run anything off of it.

**Reply by uv7jWsX7qO** [reply]
The only other thing that comes to mind is if I'm getting the IP Address of my PS2 wrong when setting that in OPL. I've tried to determine what the correct IP Address is and for now I've just set it so that it's in the same range as the pi. 169.254.etc. But given that it's in static mode, should I set that to something specific.

**Reply by uv7jWsX7qO** [reply]
UPDATE: Figured it out.

So for anyone else that is going through this. Make sure that the IP Address that you set in OPL is the same as your pi for the first 3 segments. In my case it was 169.254.101.XX (where XX could probably be anything that isn't your pi).

For MGS2 I did set Mode 2 on and now it's booting up perfectly.

**Reply by danmons** [reply]
An IP address of 169.X.X.X is when an interface has been set to DHCP (i.e.: it asks for an address) but receives none, times out, and choose a random address in the 169.254.0.1 - 169.254.255.254 range.   More information here:
https://en.wikipedia.org/wiki/Link-local_address#IPv4

It's fine to use this, however it requires the devices (in your case the Pi) to search for an address and time out, which can take a minute or so after boot.  The better way is to assign a static address to both sides in the same subnet.

**Reply by uv7jWsX7qO** [reply]
Agreed. I've since got a lot more up to speed and have since set a static address which works a lot more predictably.
